libraries/AP_Camera/AP_Camera.cpp

```diff
@@ -283,6 +290,12 @@ void AP_Camera::update()
         return;
     }
 
+#if APM_BUILD_TYPE(APM_BUILD_ArduPlane) || APM_BUILD_TYPE(APM_BUILD_APMrover2) || APM_BUILD_TYPE(APM_BUILD_ArduCopter) || APM_BUILD_TYPE(APM_BUILD_ArduSub)
```

Comment:
I missed earlier review comments, but why is this #if needed? I think if a build doesn't support mode AUTO then setting _auto_only and not getting photos is absolutely acceptable.